### PR TITLE
refactor: used requestAnimationFrame instead timeouts

### PR DESCRIPTION
**Pull request by ahilles107**
- added grunt to dev-dependencies

I'm not sure it this is what was described in #49 but requestAnimationFrame is much better and solid solution than timeout. 

Based on informations from: https://gist.github.com/Warry/4254579 and http://www.html5rocks.com/en/tutorials/speed/scrolling/ 

Fixes: #49

### CODE REVIEW

**Comment by hampelm** [comment]
Nice! I didn't know about that. Let me know when this is good to go and I'll test & merge.

**Comment by ahilles107** [comment]
Give me feedback before merge - maybe we can use it also for other function (where we listen on scroll).

File: src/carebot-tracker.js [hunk 1]
```diff
@@ -304,23 +313,35 @@
             var percent = depthPercent();
             var bucket = percentBucket(percent);
             if (bucket > previousBucket) {
-                var seconds = timer.check().seconds;
-                callback(bucket, seconds);
-            } else {
-                // The user is scrolling back up.
+                callback(bucket, timer.check().seconds);
                 previousBucket = bucket;
```
Comment (hampelm) [inline src/carebot-tracker.js]: Could you pull this line out of the if statement? The desired behavior is: 
- Report when the user scrolls down the page (working well)
- If the user scrolls back up and starts scrolling down again, the callback is called again for those buckets (requires us to set `previousBucket` no matter if the current bucket is higher or lower)

Comment (hampelm) [inline src/carebot-tracker.js]: Ideally I'll get selenium-based tests running to cover this.

Comment (ahilles107) [inline src/carebot-tracker.js]: Hey, thanks for review! But i don't understand why we should send events/log percents of page already showed for reader? You know that Google Analytics accept only 500 hits per session? You want log that user "re-read" article? Just want to understand reason ;)

Comment (hampelm) [inline src/carebot-tracker.js]: For sure. We want to cover the situation where someone skims or scrolls down quickly (eg to see how long the article is), and then re-reads. 

I've tried to keep the tracking code analytics-agnostic so people can use the stats in different ways, and don't necessarily have to re-report them to GA. For example, we're thinking about testing out a different system for tracking this event data. 

As far as the 500 event limit goes, for most articles I think we're only seeing 10-15 events from the scroll depth tracker. The bigger one is the linger time indicator. I need to add code to stop reporting once the page is inactive for a certain amount of time.

Comment (hampelm) [inline src/carebot-tracker.js]: I checked earlier today and it looks like our average events per session right now is still less than 10.

Comment (hampelm) [inline src/carebot-tracker.js]: Hi @ahilles107 -- just wanted to check in and see if you're good with pulling out the `previousBucket = bucket` line. We'd like to deploy the improvements!

Comment (ahilles107) [inline src/carebot-tracker.js]: ## Hey, I will back to it today (in next 3h).

Paweł Mikołajczuk

**Comment by ahilles107** [comment]
it's  done.

**Comment by hampelm** [comment]
:+1: Thanks!

**Pull request merged**